test/orange/g4org/SolidConverter.test.cc

```diff
@@ -195,6 +195,93 @@ TEST_F(SolidConverterTest, displaced)
         {{1, 2, 3}, {2, 2, 3}, {3, 0, 0}});
 }
 
+TEST_F(SolidConverterTest, generictrap)
+{
+    {
+        this->build_and_test(
+            G4GenericTrap("boxGenTrap",
+                          30,
+                          {{-10, -20},
+                           {-10, 20},
+                           {10, 20},
+                           {10, -20},
+                           {-10, -20},
+                           {-10, 20},
+                           {10, 20},
+                           {10, -20}}),
+            R"json({"_type":"shape","interior":{"_type":"gentrap",
+                "halfheight":3.0,
+                "lower":[[1.0,-2.0],[1.0,2.0],[-1.0,2.0],[-1.0,-2.0]],
+                "upper":[[1.0,-2.0],[1.0,2.0],[-1.0,2.0],[-1.0,-2.0]]},
+                "label":"boxGenTrap"})json",
+            {{-1, -2, -3}, {1, 2, 3}, {1, 2, 4}});
+    }
+
+    {
+        this->build_and_test(
+            G4GenericTrap("trdGenTrap",
+                          3,
+                          {{-10, -20},
+                           {-10, 20},
+                           {10, 20},
+                           {10, -20},
+                           {-5, -10},
+                           {-5, 10},
+                           {5, 10},
+                           {5, -10}}),
+            R"json({"_type":"shape","interior":{"_type":"gentrap"
+            ,"halfheight":0.30000000000000004,
+            "lower":[[1.0,-2.0],[1.0,2.0],[-1.0,2.0],[-1.0,-2.0]],
+            "upper":[[0.5,-1.0],[0.5,1.0],[-0.5,1.0],[-0.5,-1.0]]},
+            "label":"trdGenTrap"})json",
+            {{-1, -2, -4}, {-1, -2, -3}, {0.5, 1, 3}, {1, 1, 3}});
+    }
+
+    {
+        this->build_and_test(
+            G4GenericTrap("trap_GenTrap",
+                          40,
+                          {{-9, -20},
+                           {-9, 20},
+                           {11, 20},
+                           {11, -20},
+                           {-29, -40},
+                           {-29, 40},
+                           {31, 40},
+                           {31, -40}}),
+            R"json({"_type":"shape","interior":{"_type":"gentrap",
+            "halfheight":4.0,
+            "lower":[[1.1,-2.0],[1.1,2.0],[-0.9,2.0],[-0.9,-2.0]],
+            "upper":[[3.1,-4.0],[3.1,4.0],[-2.9,4.0],[-2.9,-4.0]]},
+            "label":"trap_GenTrap"})json",
+            {{-1, -2, -4 - 1.e-6}, {-1, -2, -3}, {0.5, 1, 3}, {1, 1, 3}});
+    }
+
+    // TODO: most generic gentrap with twisted side faces
+    /*
```

Comment:
For "todo" code chunks like this (they should rarely be allowed but I think this is a reasonable case), enclose them with `#if 0` and `#endif`.

Better yet would be to write `EXPECT_THROW(this->build_and..., celeritas::RuntimeError);` since for now we should get a "not implemented" when converting.

Reply:
I've added the EXPECT_THROW() suggestion.
Thanks for pointing that out.